lib/msf/core/payload/stager.rb

```diff
@@ -156,7 +156,14 @@ def handle_connection(conn, opts={})
     # If the stage should be sent over the client connection that is
     # established (which is the default), then go ahead and transmit it.
     if (stage_over_connection?)
-      p = generate_stage
+      uuid_raw = conn.get_once(16, 1)
```

Comment:
This is the bit that I really need help on. What I'm doing here is attempting to read 16 bytes off the socket immediately. If something comes down it's assumed to be the UUID. If we timeout, then we just move on.

I'm not sure this approach is ideal, though it does seem to work as part of my testing. I would love to hear ideas for a better approach, or thoughts on whether the timeout should be extended.